src/main/java/com/mercateo/spring/security/jwt/security/JWTAuthenticationTokenFilter.java

```diff
@@ -15,40 +15,64 @@
  */
 package com.mercateo.spring.security.jwt.security;
 
+import com.mercateo.spring.security.jwt.token.exception.InvalidTokenException;
+
 import java.io.IOException;
 
 import javax.servlet.FilterChain;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.security.authentication.AnonymousAuthenticationProvider;
+import org.springframework.security.authentication.AnonymousAuthenticationToken;
 import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.AuthorityUtils;
 import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import org.springframework.util.AntPathMatcher;
 
-import com.mercateo.spring.security.jwt.token.exception.InvalidTokenException;
-
-import lombok.extern.slf4j.Slf4j;
+import io.vavr.collection.Set;
 
 @Slf4j
 public class JWTAuthenticationTokenFilter extends AbstractAuthenticationProcessingFilter {
 
     private final static String TOKEN_HEADER = "authorization";
 
-    public JWTAuthenticationTokenFilter() {
+    private final AntPathMatcher antPathMatcher =  new AntPathMatcher();
+
+    @NonNull
+    private Set<String> unauthenticatedPaths;
+
```

Comment:
Bad API design. The token filter gets a set of strings. This given strings have a special meaning because they bypass the normal jwt-token handling.
Better: use a builder pattern or use a special method like addUnauthenticatedPaths. You must make the exceptional cases explicitly!